microprofile/server/src/main/java/io/helidon/microprofile/server/RoutingBuilders.java

```diff
@@ -0,0 +1,108 @@
+
+import java.util.Optional;
+
+import io.helidon.config.Config;
+import io.helidon.microprofile.server.spi.MpServiceContext;
+import io.helidon.webserver.Routing;
+
+/**
+ * Provides {@link Routing.Builder} instances (for the default and the actual)
+ * for a Helidon MP service, based on configuration for the component (if any)
+ * and defaults otherwise.
+ */
+public interface RoutingBuilders {
+
+    /**
+     *
+     * @return the default {@code Routing.Builder} for the component
+     */
+    Routing.Builder defaultBuilder();
+
+    /**
+     *
+     * @return the actual {@code Routing.Builder} for the component; might be the default
+     */
+    Routing.Builder builder();
+
+    /**
+     * Prepares the default and actual {@link Routing.Builder} instances based
+     * on the "routing" configuration for the specific component.
+     *
+     * @param context the {@code MpServiceContext} for the calling service
+     * @param componentName config key under which "routing" config might exist for the component of interest
+     * @return {@code RoutingBuilders} based on the named config (or default)
+     */
+    static RoutingBuilders newRoutingBuilders(MpServiceContext context, String componentName) {
+        return newRoutingBuilders(context, context.helidonConfig().get(componentName));
+    }
+
+    /**
+     * Prepares the default and actual {@link Routing.Builder} instances based
+     * on the "routing" configuration for the specific component configuration.
+     *
+     * @param context the {@code MpServiceContext} for the calling service
+     * @param componentConfig the configuration for the calling service
+     * @return {@code RoutingBuilders} based on the config (or default)
+     */
+    static RoutingBuilders newRoutingBuilders(MpServiceContext context, Config componentConfig) {
+        final Routing.Builder defaultRoutingBuilder = context.serverRoutingBuilder();
+        final Routing.Builder actualRoutingBuilder
+                = componentConfig.get("routing")
+                        .asString()
+                        .flatMap(routeName -> {
+                            // support for overriding the routing back to default port using config
+                            if ("@default".equals(routeName)) {
+                                return Optional.empty();
+                            } else {
+                                return Optional.of(routeName);
+                            }
+                        })
+                        // use named routing
+                        .map(context::serverNamedRoutingBuilder)
+                        // use default server routing
+                        .orElse(defaultRoutingBuilder);
+        return new RoutingBuildersImpl(defaultRoutingBuilder, actualRoutingBuilder);
+    }
+
+    /**
+     * Package-private implementation of the {@code RoutingBuilders} interface.
+     */
+    class RoutingBuildersImpl implements RoutingBuilders {
```

Comment:
As this class is defined within the interface, it is implicitly public. You need to move it to the package to make it package private.

Reply:
Refactored.